pandas/core/indexing.py

```diff
@@ -1854,7 +1859,7 @@ def _setitem_with_indexer_missing(self, indexer, value):
             if index.is_unique:
                 new_indexer = index.get_indexer([new_index[-1]])
                 if (new_indexer != -1).any():
-                    return self._setitem_with_indexer(new_indexer, value)
+                    return self._setitem_with_indexer(new_indexer, value, "loc")
```

Comment:
comment explaining that we only get here with name=="loc" and thats why its hardcoded (or assertion)